### ISSUE TITLE
Unable to save email address as username (since updating from 1.8.3)

### ISSUE THREAD

**Issue by gurtjun** [Dielee/volvo2mqtt#97]
**Describe the bug**
Since updating to the latest version (from 1.8.3 to 1.8.6) the plugin is not working anymore for me. It seems that I cannot save my email as username anymore... Was there a change in the regular expression? My email is something like `jo-hn.doe@example.com`

I get the following error: 
```
Failed to save add-on configuration, does not match regular expression ^([\w+\.]+@([\w-]+\.)+[\w-]{2,4})|(\+\d{5,20})$.
```

**To Reproduce**:<br>
1. Go to the add-on
2. Click on 'Configuration'
3. Fill in all options (see my config below)
5. Click 'Save'
6. See error

Update Interval:
`300`

Babel Locale:
`nl_BE`

Timezone:
`Europe/Brussels`

MQTT Broker settings:
```yaml
broker: core-mosquitto
port: auto_port
username: xxxx
password: xxxx
```

Volvo Configuration Options:
```yaml
username: jo-hn.doe@example.com
password: xxxx
vin: xxxx
vccapikey:
  - xxxx
odometerMultiplier: 1
averageSpeedDivider: 1
averageFuelConsumptionMultiplier: 1
```

**Expected behavior**:<br>
I expect the add-on to work like before, I was able to load all the details of my car before updating to the latest version

**Screenshots**:<br>

**Version info**:<br>
Home Assistant: `2023.8.4`
Supervisor: `2023.08.1`
Operating System: `10.5`
Add-on: `1.8.6`

**Comment by Dielee** [comment]
I need your real email address to check the ReEx validation.

**Comment by gurtjun** [comment]
> I need your real email address to check the ReEx validation. Is this email address nearly the original ? `jo-hn.doe@example.com`

Yes, it includes the `-` and the `.`

**Comment by gurtjun** [comment]
created #98 with a possible fix

**Comment by Dielee** [comment]
Should be fixed with version v1.8.7!. Thanks for your PR.

**Comment by gurtjun** [comment]
Yes, I'm able to save my configuration again! Thanks!
